File: javascript/two_pointers/228. Summary Ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    if (nums.length === 0) return [];
    
    res = [];
    let l = 0, r = 0;
    while (r < nums.length) {
        if (r < nums.length-1 && nums[r]+1 == nums[r+1]) {
            r++;
        } else {
            nums[l] == nums[r] ? res.push("" + nums[l]) : res.push("" + nums[l] + "->" + nums[r]);
            l = r + 1;
            r++;
        }
    }
    return res;
};

File: python/two_pointers/228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if len(nums) == 0:
            return []

        res = []
        l = r = 0
        while r < len(nums):
            if r < len(nums) - 1 and nums[r] + 1 == nums[r+1]:
                r += 1
            else:
                if nums[l] == nums[r]:
                    res.append(str(nums[l]))
                else:
                    res.append(str(nums[l]) + "->" + str(nums[r]))
                l = r = r + 1

        return res